docs/workflows/react-native-specific-workflow-steps/react-native-unit-test.md

```diff
@@ -0,0 +1,67 @@
+
+For detailed information for continuous testing. Please visit our [React Native Continuous Testing documentation](/continuous-testing/react-native-testing/react-native-unit-test-with-jest).
+
+### Prerequisites
+
+The workflow steps that need to be executed before running the **React Native Unit Test** workflow step, along with their respective reasons, are listed in the table below.
+
+| Prerequisite Workflow Step                                                                 | Description                                                                                                                                                                            |
+|--------------------------------------------------------------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| [**Git Clone**](/workflows/common-workflow-steps#git-clone)                                | Clone the selected repository to the build machine.                                                                                                                                    |
+| [**Install Node**](/workflows/react-native-specific-workflow-steps#install-node)           | This step will install Node modules for your application.                                                                                                                              |
+| [**NPM/Yarn Commands**](/workflows/react-native-specific-workflow-steps/npm-yarn-commands) | This step install the [NPM](https://www.npmjs.com/) or [Yarn](https://www.npmjs.com/package/yarn) package manager to install specific dependencies for your React Native applications. |
+
+<Screenshot url='https://cdn.appcircle.io/docs/assets/BE4443-rnUnitFlowOrder.png'/>
+
+:::danger Step Rule
+
+There's one important setting that you should change. If any workflow steps fail, Appcircle automatically skips other steps and jumps directly to the **Export Build Artifacts** step. However, it is possible that some of your tests may fail. **If React Native Unit Test doesn't run, not all tests will be completed and test results will not be generated.** You should turn on the following toggles so that **React Native Unit Test** component always runs whether your tests fail or pass.
+
+- Always run this step even if the previous steps fail to **ON**
+- Continue with the next step even if this step fails to **ON**
+
+<Screenshot url='https://cdn.appcircle.io/docs/assets/unitRule.png' />
```

Comment:
Are we sure about this caution box? For example, is it reasonable to "always run unit tests" when, for example, the "npm install" step fails?

As far as I see from your test build profile, you also did not obey your caution.  😅 

<img width="1018" alt="Screenshot 2024-11-02 at 08 34 19" src="https://github.com/user-attachments/assets/406bd65d-453a-4dbb-bedf-0520a4a203cc">

IMO, confusion happened here. Test reports must run when unit/UI tests fail, but the unit/UI test step has different constraints.

Reply:
Yes you are right, I updated the screenshot and rule according to this. You can check the related screen shots for both comps